### PR TITLE
Babel 7 upgrade

### PR DESCRIPTION
**Pull request by bj00rn**
Upgrade to Babel 7

Resolves: #146 #132 #122

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -155,7 +156,7 @@ function transform(filePath, options = {}, {multiplePasses = false} = {}) {
         return [plugin, {
             ...BASE_OPTIONS,
             ...options,
-        }];
+        }, uuidv1()];
```
Comment (bj00rn) [inline test/index.js]: somewhat dirty fix, need to add random string suffix to allow plugin to run more than once

**Comment by ykzts** [comment]
https://github.com/yahoo/babel-plugin-react-intl/blob/v2.4.0/.travis.yml#L4-L5

Babel v7 is no longer supported under Node.js v5.

File: test/index.js (hunk 1 above)
Comment (bj00rn) [inline test/index.js]: somewhat dirty fix, need to add random string suffix to allow plugin to run more than once

**Comment by tstirrat15** [comment]
Two things: one is that Babel 7.0.0 (i.e. non-beta) is out. Is it worth upgrading directly to that?

The other is that I'd love to see this work go through. What's the status? Is there any way that I can help it along?

**Comment by bj00rn** [comment]
@tstirrat15 moved to 7.0

**Comment by bj00rn** [comment]
@ericf what are the chances of having this merged?

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 sudo: false
 language: node_js
 node_js:
-    - 4.2
-    - 5
     - 6
```
Comment (okuryu) [inline .travis.yml]: I think we should have testing on v8 and v10 right now.

Comment (bj00rn) [reply .travis.yml]: seems a little out-of-scope for this PR. maybe a separate PR for this?

File: package.json [hunk 3]
```diff
@@ -10,29 +10,33 @@
   "main": "lib/index.js",
   "author": "Eric Ferraiuolo <edf@ericf.me>",
   "dependencies": {
-    "babel-runtime": "^6.2.0",
+    "@babel/runtime": "7.0.0",
```
Comment (okuryu) [inline package.json]: Why it's not semver style like `^7.0.0`? any concern?

Comment (bj00rn) [reply package.json]: fixed. thanks

didn't notice. i just ran babel-upgrade https://github.com/babel/babel-upgrade

File: src/index.js [hunk 4]
```diff
@@ -4,7 +4,7 @@
  * See the accompanying LICENSE file for terms.
  */
 
-import * as p from 'path';
```
Comment (serwer-WittGruppe) [inline src/index.js]: Issue seems to be here where upath is clearly used in library code, but not added to normal dependencies but only devDependencies.

**Pull request merged**